QuoteEngine.py

```diff
@@ -0,0 +1,30 @@
+import requests
+import yaml
+import random
+
+
+class QuoteEngine:
+
+    def __init__(self, logger):
+        self.quote_url = "https://cdn.rawgit.com/tulul/tulul-quotes/master/quote.yaml"  # noqa
+        # Note: rawgit does not have 100% uptime, but at
+        # least they're not throttling us.
+
+        self.logger = logger
+        self.cache = []
+        self.default_empty_error = "Nothing in cache. Forget to refresh_cache?"
+
+    def retrieve_random(self):
+        cache = self.cache
+        index = random.randrange(0, len(cache))
+        return self.format_quote(cache[index])
+
+    def format_quote(self, q):
+        return "%s - %s, %s" % (q['quote'], q['author'], q['author_bio'])
+
+    def refresh_cache(self):
+        content = requests.get(self.quote_url).content
```

Comment:
Any particular reason to use `.content` instead of `.text`? `.content` returns bytes, and from PyYAML docs I think the `.load` function expects strings. I know bytes and strings are interchangeable in some cases but it's better to be explicit don't you think?